yargs.js

```diff
@@ -993,27 +987,31 @@ function Yargs (processArgs, cwd, parentRequire) {
     }
 
     // If the help or version options where used and exitProcess is false,
-    // or if explicitly skipped, we won't run validations
+    // or if explicitly skipped, we won't run validations.
     if (!skipValidation) {
       if (parsed.error) throw new YError(parsed.error.message)
 
       // if we're executed via bash completion, don't
       // bother with validation.
       if (!argv[completion.completionKey]) {
-        validation.nonOptionCount(argv)
-        validation.missingArgumentValue(argv)
-        validation.requiredArguments(argv)
-        if (strict) validation.unknownArguments(argv, aliases)
-        validation.customChecks(argv, aliases)
-        validation.limitedChoices(argv)
-        validation.implications(argv)
-        validation.conflicting(argv)
+        self._runValidation(argv, aliases)
       }
     }
 
     return setPlaceholderKeys(argv)
   }
 
+  self._runValidation = function (argv, aliases) {
```

Comment:
`_runValidation` was pulled into a helper, to make it easier to run optionally -- this was so that we could hold off on running it until the last moment in the case of commands.